backend/src/main/java/gov/cdc/usds/simplereport/db/model/ApiAuditEvent.java

```diff
@@ -0,0 +1,145 @@
+import org.hibernate.annotations.Type;
+
+/**
+ * Model to meet the following specification:
+ *
+ * <h3>AU-3</h3>
+ *
+ * We will capture in our database, for each API request by an authenticated user, the following
+ * data:
+ * <li>the rest endpoint or graphql query, with parameters (partial—not clear what we need for REST)
+ * <li>the identity of the user making the request
+ * <li>the permissions applicable to that user when they made the request
+ * <li>the identity of the tenant organization whose data was being queried or altered
+ * <li>whether the action was permitted or was denied
+ * <li>the origin of the HTTP request
+ * <li>the date and time of the request
+ */
+@Entity
+@Immutable
+public class ApiAuditEvent {
+
+  /** The primary key for the log entry, which can certainly be a random UUID */
+  @Id
+  @Column // generate locally or on the server? on the server doesn't batch well, but... so?
+  @GeneratedValue(generator = "UUID4")
+  private UUID id;
+
+  /**
+   * The timestamp for the log event creation (this is likely the *end* of request processing, so
+   * trying to match on may get dicey, and we may need to add another field for that.
+   */
+  @Column(updatable = false)
+  @Temporal(TemporalType.TIMESTAMP)
+  @CreationTimestamp
+  private Date eventTimestamp;
+
+  /** The correlation ID from server logs */
+  @Column private String requestId;
+
+  @Column
+  @Type(type = "jsonb")
+  private HttpRequestDetails httpRequestDetails;
+
+  @Column
+  @Type(type = "jsonb")
+  private GraphQlInputs graphqlQueryDetails;
+
+  @Column
+  @Type(type = "list-array")
+  private List<String> graphqlErrorPaths;
+
+  @ManyToOne(optional = true)
+  @JoinColumn(name = "api_user_id")
+  private ApiUser user;
+
+  @ManyToOne(optional = true)
+  @JoinColumn(name = "organization_id")
+  private Organization organization;
+
+  @Column
+  @Type(
+      type = "list-array",
+      parameters = {@Parameter(name = ListArrayType.SQL_ARRAY_TYPE, value = "text")})
+  @Enumerated(EnumType.STRING)
+  private List<UserPermission> userPermissions;
+
+  @Column private boolean isAdminUser; // don't love this but it is actually pretty specific
+
+  @ManyToOne(optional = true)
+  @JoinColumn(name = "patient_link_id")
+  private PatientLink patientLink;
+
+  public ApiAuditEvent(
+      String requestId,
+      HttpRequestDetails httpRequestDetails,
+      GraphQlInputs graphqlQueryDetails,
+      List<String> errorPaths,
+      ApiUser apiUser,
+      List<UserPermission> permissions,
+      boolean isAdmin,
+      Organization org) {
+    super();
+    this.requestId = requestId;
+    this.httpRequestDetails = httpRequestDetails;
+    this.graphqlQueryDetails = graphqlQueryDetails;
+    this.graphqlErrorPaths = errorPaths;
+    this.user = apiUser;
+    this.organization = org;
+    this.isAdminUser = isAdmin;
+    ArrayList<UserPermission> permissionCopy = new ArrayList<>(permissions);
+    permissionCopy.sort(null);
```

Comment:
It the motive for sorting just to make compares easier at some later point?

Reply:
Exactly. However, thinking about that I realize that this field is a nasty trap for if and when we delete any UserPermission values. Probably better store it as String, and sort it stringwise?